### PR TITLE
Add ASt NeuroPixels dataset

### PR DESCRIPTION
**Pull request by weiglszonja**
NeuroPixels dataset consists of:
- recording in SpikeGLX format
- sorting in Phy format
- Histology image: shows where the neuropixel probes were inserted

### CODE REVIEW

File: src/tye_lab_to_nwb/ast_neuropixels/convert_session.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        For testing purposes, when stub_test=True only writes a subset of ecephys data.
+        Default is to write the whole ecephys recording to the file.
+    """
+
+    source_data = dict()
+    conversion_options = dict()
+
+    # Add Recording
+    ecephys_recording_folder_path = Path(ecephys_recording_folder_path)
+    spike_glx_ap_bin_file_paths = list(ecephys_recording_folder_path.glob("*.ap.bin"))
```
Comment (weiglszonja) [inline src/tye_lab_to_nwb/ast_neuropixels/convert_session.py:46]: However #23 if they collected more than 1 subject data within the same folder then the folder approach wouldn't help, or based on subject metadata (from the Excel file) I could look up which "`imec{ind}.ap.bin`" to pick from the folder.

Comment (weiglszonja) [reply src/tye_lab_to_nwb/ast_neuropixels/convert_session.py]: Changed source data to file path https://github.com/catalystneuro/tye-lab-to-nwb/pull/24/commits/15fbaccc476d167ffe9cf673fadf8b12855b5ce8

**Comment by CodyCBakerPhD** [comment]
Overall looks good, my only requested changes involve the massive number of properties added to the `Units` table

Could you do 2 things there

1) Expand the column names to use full English references; _e.g._, `isi_viol` -> `isi_violation`
2) Add descriptions to many of the columns; you can use [this reference](https://github.com/catalystneuro/IBL-to-nwb/blob/main/ibl_to_nwb/updated_conversion/metadata/ecephys.yml#L35) to help

**Comment by weiglszonja** [comment]
Thank you @CodyCBakerPhD!
For reference the list of items I'll need to ask descriptions for:

3 types of "amplitude" columns that have different values:
```
Amplitude -> the values for the first 3 units: [2967.  4477.  1957.8]
amp -> the values for the first 3 units: [179.66105769 383.97253788 124.73197993]
amplitude -> the values for the first 3 units: e.g. [61.62191772 88.24165344 71.04180145]
amplitude_cutoff 
```
Other columns that I'm not sure about whay they mean:
```
PT_ratio
d_prime
depth
duration
epoch_name
epoch_name_quality_metrics
epoch_name_waveform_metrics
halfwidth
l_ratio
sh -> all zeros
spread
velocity_above
velocity_below
```

**Comment by CodyCBakerPhD** [comment]
@weiglszonja Minor conflict here; I'd try to resolve but I don't want to accidentally mess anything up

**Comment by weiglszonja** [comment]
thank you @CodyCBakerPhD , I updated this.

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**